### PR TITLE
[Security] Use UserBadge in all passports

### LINKED ISSUE #15112: [Security] Remove deprecated support for passing a UserInterface implem…
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#40487
| PR author(s) | @wouterj
| Merged in    | 5.3-dev

### PR DESCRIPTION
**Pull request by wouterj**
Fixes #15112

### CODE REVIEW

**Comment by OskarStark** [comment]
As the corresponding PR is merged into `5.x` aka `5.3`, should this PR target `5.3` instead?

**Comment by wouterj** [comment]
This feature is supported since 5.2 (and deprecated in 5.2), and removed in 5.3. I only found out that we didn't correctly update all examples after deleting the feature :). So 5.2 is the correct target

File: security/experimental_authenticators.rst [hunk 1]
```diff
@@ -494,15 +494,15 @@ The following credential classes are supported by default:
         use Symfony\Component\Security\Http\Authenticator\Passport\Credentials\PasswordCredentials;
 
         // ...
-        return new Passport($user, new PasswordCredentials($plaintextPassword));
+        return new Passport(new UserBadge($email), new PasswordCredentials($plaintextPassword));
```
Comment (OskarStark) [inline security/experimental_authenticators.rst:497]: Can you please add a use statement? Thanks

Comment (wouterj) [reply security/experimental_authenticators.rst:497]: I don't think it's needed. This is a really small code snippet inside a definition list (so I prefer to keep it as minimal as possible). The use statement is present in all code examples prior to this change (but those were already correct).

**Review by OskarStark — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Thanks Wouter.

**Pull request merged**